nixos/modules/misc/mandoc.nix

```
@@ -5,59 +5,191 @@ let
 
   cfg = config.documentation.man.mandoc;
 
-in {
+  toMandocOutput = output: lib.concatLines (
+    lib.mapAttrsToList
+      (
+        name: value:
+          if lib.isString value then "output ${name} ${value}"
+          else if lib.isInt value || lib.isPath value then "output ${name} ${builtins.toString value}"
```

`lib.types.path` can either be a path value or a string that contains a (absolute) path (so not equivalent to `isPath`). For path values, we want to use string interpolation, so that they are imported into the nix store and always available as long as the corresponding system configuration is available. (`toString` just converts the path to a corresponding string without importing it into the store.) For stringly typed paths, we can't really do better, but it is a general convention that paths in strings are used for impure paths in nixos configurations.

```suggestion
          if lib.isString value || lib.isPath value then "output ${name} ${value}"
          else if lib.isInt value then "output ${name} ${builtins.toString value}"
```

---

Oh, thanks! I happened to encounter a similar problem two days ago, and I thought I needed to use some kinds of built-in functions to convert it.